bin/specextract

```diff
@@ -937,56 +937,31 @@ def check_event_stats(file,refcoord,weights_check):
 
 def create_hrc_resp(specfile,rmf_file,refcoord,full_outroot,asp_param,clobber,verbose,mskfile,skyx,skyy,instrument,chip_id):
 
-    # check number of channels in PI file to determine RMF file to use from CalDB
-    dmlist.punlearn()
-    dmlist.infile = specfile
-    dmlist.opt = "counts"
-
-    chan = int(dmlist())
-
-    detector = get_keyval(specfile,"detnam")
-
     if rmf_file.upper() == "CALDB":
-
         # cf. https://cxc.cfa.harvard.edu/cal/Hrc/detailed_info.html#rmf for HRC RMF information
-        #
-        # get path from $CALDB environmental variable
-        CALDB = os.environ["CALDB"]
-
-        if chan == 1024:
-            # use SAMP responses
-            if detector.upper() == "HRC-S":
-                rmf = "{}/data/chandra/hrc/rmf/hrcsD1999-07-22samprmfN0001.fits".format(CALDB)
-            elif detector.upper() == "HRC-I":
-                rmf = "{}/data/chandra/hrc/rmf/hrciD1999-07-22samprmfN0001.fits".format(CALDB)
-            else:
-                raise ValueError("{0} has an invalid HRC detnam value.".format(specfile))
-
-        elif chan == 256:
-            # use non-SAMP responses
-            if detector.upper() == "HRC-S":
-                rmf = "{}/data/chandra/hrc/rmf/hrcsD1999-07-22rmfN0001.fits".format(CALDB)
-            elif detector.upper() == "HRC-I":
-                rmf = "{}/data/chandra/hrc/rmf/hrciD1999-07-22rmfN0002.fits".format(CALDB)
-            else:
-                raise ValueError("{0} has an invalid HRC detnam value.".format(specfile))
+ 
+        calquiz.punlearn()
+        calquiz.infile = specfile
+        calquiz.product = "MATRIX"
+        calquiz.calfile = "CALDB"
+        # calquiz.infile = "none"
+        # calquiz.telescope = "chandra"
+        # calquiz.instrument = "hrc"
+        # calquiz.calfile = "CALDB(DETNAM={})".format(get_keyval(specfile,"detnam"))
+
+        with suppress_stdout_stderr():
+            calquiz()
+
+        if calquiz.outfile == None:
+            raise IOError("{0} is an invalid HRC spectral file, as non-SAMP responses are no longer supported by the CALDB.  Please reprocess the dataset or re-download the observation from the archive.".format(specfile))
 
         else:
```

Comment:
Something my python liter is complaining about in my code is the use of

    if condition:
        raise SomeError()
    else
        do_something()

since it wants you to avoid the "redundant" `else` and write

    if condition:
        raise SomeError()
    
    do_something()

I think that makes sense to do here.